src/webhooks/types/directory.rs

```diff
@@ -0,0 +1,137 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    directory_sync::DirectoryType, organizations::OrganizationId, KnownOrUnknown, Timestamps,
+};
+
+/// The ID of a [`Directory`].
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct DirectoryId(String);
+
+impl Display for DirectoryId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<String> for DirectoryId {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+impl From<&str> for DirectoryId {
+    fn from(value: &str) -> Self {
+        Self(value.to_string())
+    }
+}
+
+/// The state of a [`Directory`].
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum DirectoryState {
+    /// The directory is linked.
+    Linked,
+
+    /// The directory is unlinked.
+    Unlinked,
```

Comment:
The `state` in the directory webhooks will be `active` / `inactive` instead of `linked` / `unlinked`:

```suggestion
    Inactive,
```